server/interceptor/interceptor.go

```diff
@@ -231,7 +231,7 @@ type rpcInfo struct {
 	treeID int64
 
 	// opts is the trees.GetOpts appropriate to this RPC (TreeType, readonly vs readwrite, etc).
-	// opts is not set if doesNotHaveTree is true.
+	// opts is not set if doNotIntercept is true.
```

Comment:
Actually,  things would be easier to follow after the restructure if a couple more comments were updated here.  IIUC

 - `doNotIntercept` could have a comment to say "if this is set, none of the other fields in the structure are set" (and then drop the new sentence for `opts`, as it will be redundant)
 - Need clarification of whether `treeID` can still be zero (when `doNotIntercept==false`)
    - If it can be, then is `opts` valid in that case (i.e. should the change have been `doesNotHaveTree` => `treeID==0`?)
    - If not (which I *think* is now the case), then there's probably a missing error case in `Before()`

Reply:
Would you mind if punted those to a separate PR?

(Sorry, I didn't see this comment before the last reply.)

Reply:
OK, just leave a TODO then.